test/integration/auth.js

```diff
@@ -70,22 +75,47 @@ function test(utils) {
   }
 
   /**
+   * Runs cleanup routine that could affect outcome of tests and removes any
+   * intermediate users created.
+   * @param {string} label The test label.
    * @return {Promise} A promise that resolves when test preparations are ready.
    */
-  function before() {
+  function cleanup(label) {
     // Delete any existing users that could affect the test outcome.
     var promises = [
       deletePhoneNumberUser(testPhoneNumber),
       deletePhoneNumberUser(testPhoneNumber2),
       deletePhoneNumberUser(nonexistentPhoneNumber),
       deletePhoneNumberUser(updatedPhone)
     ];
+    // Delete list of users for testing listUsers.
+    uids.forEach(function(uid) {
+      promises.push(
+        admin.auth().deleteUser(uid)
+          .catch(function(error) {
+            // Suppress user not found error.
+            if (error.code !== 'auth/user-not-found') {
+              throw error;
+            }
+          })
+      );
+    });
     return Promise.all(promises)
       .catch(function(error) {
-        utils.logFailure('before()', error);
+        utils.logFailure(label + '()', error);
       });
   }
 
+  /** Clean up routine to run before tests. */
+  function before() {
+    return cleanup('before');
+  }
+
+  /** Clean up routine to run after tests. */
+  function after() {
+    return cleanup('after');
```

Comment:
It's not clear to me why this needs to run both before and after. Please add a comment, if it's for legit reason.

Reply:
Done.